### ISSUE TITLE
Switch statement formatting

### ISSUE THREAD

**Issue by DavidBrainard** [isetbio/isetbio#279]
Where do we want comments for each case line?  It initially seemed natural to me to put them above the case statement, but then when you do Matlab's autoformat they get indented in relative to the case statement they apply to.  If you put them just below the case statement, then the autoformat still indents them, but to me this looks reasonable.  So I have taken to putting them below.

In reviewing the wavefront code, I am finding both conventions within the same routine.  I think we should pick one.  I prefer below, but as long as we do it consistently I will be happy.  Thoughts?

**Comment by DavidBrainard** [comment]
I also think I'd avoid %% for the case comments, as they aren't really cells.  This too varies within the code.

**Comment by wandell** [comment]
First world problems 😉

Below when the comment is about that case.  Sometimes, however, the comment applies to a group of case statements.  Less sure then.  Sometimes I fight the Matlab formatting, hence what you see.

Brian

--------------------
Sent from my iPad
________________________________

**Comment by render-toolbox** [comment]
Agreed, minor -- I am not going to change these systematically.  But when I see things I would like to nudge them in a good direction that we agree on.

One approach to comments that apply to multiple cases is to put them above the switch, and make clear which cases the comment refers to.

I too sometimes fight Matlab's formatting.  But I always lose, because sooner or later someone types cmd-A cmd-I and it's all over.

DB

**Comment by DavidBrainard** [comment]
In looking a little more at the group comments, I see the issue.  Not sure how to best group things in a switch.
Could break down code into multiple switches, but that makes it harder to add "otherwise" checks that
complain if there is no match to any of the cases.  I guess you could set a no match flag to false
before the first switch and somehow try to track it down through everything.  Will ponder.

**Comment by DavidBrainard** [comment]
I implemented the separation of the big switch in oiPlot down into multiple smaller switches, each grouped by theme.  Each switch unsets a flag if none of its cases matched the passed plot type, and at the end there is a conditional that throws an error if the passed plot type was not handled in any switch.

Doing this allows a more pleasing commenting style, as each switch has a introductory comment at the beginning, and then the cases have their specific comments.

Not sure whether this is worth doing systematically, but it can serve as a template if we take this on in other routines.

Closing this issue, as I don't think there is a specific action item anymore.
